src/WebJobs.Script.WebHost/WebHooks/WebHookReceiverManager.cs

```diff
@@ -80,7 +80,16 @@ public WebHookReceiverManager(SecretManager secretManager)
             await request.Content.LoadIntoBufferAsync();
 
             string receiverId = function.Name.ToLowerInvariant();
-            return await receiver.ReceiveAsync(receiverId, context, request);
+
+            // Get an optional client ID. This information is passed as the receiver ID, allowing
+            // the receiver config to map configuration based on the client ID (primarily used for secret resolution).
+            string clientId = request.GetQueryNameValuePairs()
+                .FirstOrDefault(q => string.Compare(q.Key, "clientId", StringComparison.OrdinalIgnoreCase) == 0)
```

Comment:
@mathewc we talked about changing this but because of the way routes are currently mapped to functions, we won't be able to pass an additional path segment (to stay consistent with the model supported by the WebHooks SDK) and continue to route to the same function (routes are currently keyed on the full path) so our options are:

1. Stick with this model and also support path once the routing stuff is committed
2. Remove this and NOT support named keys for WebHooks until the routing stuff is committed

Thoughts?
/cc @brettsam